### PR TITLE
Add postsubmit configuration for components

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Added postsubmit jobs for: `kyma/ui-api-layer`, `console/brokers`, `console/content`, `console/catalog`, and `console/instances`
- Renamed `pipeline.sh` to `build.sh`
- Removed `SOURCES_DIR` env
- Changed tags for postsubmit jobs (Jenkins compatibility)
- Changed tags for buildpacks build for PR
- Added tests for jobs
- Updated migration guideline

> **NOTE:** The `pipeline.sh` script will be removed in separate PR for backward compatibility.

**Related issue(s)**

#58

### CODE REVIEW

File: docs/prow/migration-guide.md [hunk 1]
```diff
@@ -36,8 +39,8 @@ For example, to define a job for the `binding-usage-controller` component from t
 
 > **NOTE:** All `yaml` files in the whole `jobs` structure need to have unique names.
 
-In your job, call a `pipeline.sh` script from the buildpack image provided in the `test-infra` repository. This script executes the `Makefile` target defined for your component.
-The `pipeline.sh` script requires the **SORUCES_DIR** environment variable which is the path to the directory where the `Makefile` for your component is defined.
+In your job, call a `build.sh` script from the buildpack image provided in the `test-infra` repository. This script executes the `Makefile` target defined for your component.
+The `build.sh` script requires the path to the directory where the `Makefile` for your component as argument.
```
Comment (aszecowka) [inline docs/prow/migration-guide.md]: something is wrong with this sentence.

File: docs/prow/kyma-release-process.md [hunk 2]
```diff
@@ -105,31 +116,32 @@ job_labels_template: &job_labels_template
...
-    run_if_changed: "(components/ui-api-layer/|resources/core/values.yaml)"
-    branches:
-    - '^release-\d+\.\d+$'
-    labels:
-      preset-build-release: "true"
-      <<: *job_labels_template
-    <<: *job_template
-
+    - name: &name prow/kyma/components/ui-api-layer
+      context: *name
```
Comment (suleymanakbas91) [inline docs/prow/kyma-release-process.md]: You can remove `context` as it is `name` by default.

File: docs/prow/migration-guide.md [hunk 3]
```diff
@@ -67,39 +70,38 @@ presubmits:
...
+| **spec.containers.command** | Buildpacks have the `build.sh` script that allows you to easily trigger the build for your component. The `build.sh` script populates some environment variables, such as **DOCKER_TAG**. It also initializes Docker and executes one of the `Makefile` targets, depending on the value of the **BUILD_TYPE** variable. This environment variable can be injected by one of the build's Presets, which are **preset-build-pr**, **preset-build-release**, or **preset-build-master**. |
+| **spec.containers.args**    | The `build.sh` script requires path to the directory with the location of the `Makefile` for your component.                                                                                                                                                                                                                                                                                                                                                                          |
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply docs/prow/migration-guide.md]: The resolution of @kazydek suggestion looks weird to me. Can you double check?

Comment (Tomasz-Smelcerz-SAP) [inline docs/prow/migration-guide.md]: Something is wrong with this sentence, looks like @kazydek suggestion is not applied.

**Pull request merged**